canonical-web-and-design/maas-ui
# feat(ui): add buttons to add a bond or bridge

## Done

- Add buttons to the network tab to create a bond or bridge provided the correct selection is made.

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- View the network tab for a machine.
- Add an alias if there isn't one already.
- Hover the bond button follow the directions on the tooltip to create a bond.
- Do the same for the bridge button.
- Select different combinations of interfaces and see that you get different tooltips with hints on both buttons.

## Fixes

Fixes: #2308.
